src/main/java/ladder/domain/ResultUsers.java

```diff
@@ -0,0 +1,44 @@
+package ladder.domain;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ResultUsers {
+    private final Map<User, String> result;
+
+    public ResultUsers(Map<User, String> result) {
+        this.result = result;
+    }
+
+    public static ResultUsers of(Ladder ladder, ResultInput resultInput) {
+        List<String> userNames = ladder.getUserNames();
+        List<Integer> positions = ladder.climbAll();
+        if (userNames.size() != resultInput.size()) {
+            throw new IllegalArgumentException("사용자 수와 결과 수가 일치하지 않습니다.");
+        }
+        return new ResultUsers(
+                IntStream.range(0, userNames.size())
+                        .boxed()
+                        .collect(Collectors.toMap(
+                                i -> new User(userNames.get(i)),
+                                i -> resultInput.getResultByPosition(positions.get(i))
+                        ))
+        );
```

Comment:
여기도 IntStream 대신 userNames를 활용한 stream 가능할 것 같아요 
```suggestion
        return userNames.stream()
                .collect(Collectors.collectingAndThen(
                        Collectors.toMap(
                                User::new,
                                userName -> resultInput.getResultByPosition(positions.get(userNames.indexOf(userName)))
                        ),
                        ResultUsers::new
                ));
```